### ISSUE TITLE
No RPM with working encoder

### ISSUE THREAD

**Issue by MicrowavedBread** [clough42/electronic-leadscrew#230]
encoder in use: https://www.ebay.com/itm/274651353872
I have tried 2 of the listed encoders.

Encoder resolution: 4096
encoder pin: EQEP1
Correctly wired.

Multimeter shows 5v pulse when rotated by hand.
I see no rpm change when I rotate the encoder while plugged in.
I can send pictures of wiring if needed.
Configuration.h included below

[Configuration.txt](https://github.com/clough42/electronic-leadscrew/files/9784810/Configuration.txt)

Any and all help is much appreciated.

**Comment by jojotoshio** [comment]
> encoder in use: https://www.ebay.com/itm/274651353872 I have tried 2 of the listed encoders.
> 
> Encoder resolution: 4096 encoder pin: EQEP1 Correctly wired.
> 
> Multimeter shows 5v pulse when rotated by hand. I see no rpm change when I rotate the encoder while plugged in. I can send pictures of wiring if needed. Configuration.h included below
> 
> [Configuration.txt](https://github.com/clough42/electronic-leadscrew/files/9784810/Configuration.txt)
> 
> Any and all help is much appreciated.

Hi, I'm having a similar problem though I think I may have a bad TI board. How did you test your encoder? I'd like to test mine  as well.

**Comment by Chucksbp** [comment]
There is a lengthy discussion of this problem below, discussion 197. Many of us have had this problem. Did it work and then stop or did it never work. If it never worked id first look at and check connections as in the trouble shooting instructions then if problem is not resolved i would re program the board using the second input of the TI board. That has solved the issues we had where it worked and stopped.

**Comment by MicrowavedBread** [comment]
> 

It has never worked. And I've tried the 2nd encoder port as well will both encoders.
What troubleshooting instructions?

**Comment by MicrowavedBread** [comment]
> > encoder in use: https://www.ebay.com/itm/274651353872 I have tried 2 of the listed encoders.
> > Encoder resolution: 4096 encoder pin: EQEP1 Correctly wired.
> > Multimeter shows 5v pulse when rotated by hand. I see no rpm change when I rotate the encoder while plugged in. I can send pictures of wiring if needed. Configuration.h included below
> > [Configuration.txt](https://github.com/clough42/electronic-leadscrew/files/9784810/Configuration.txt)
> > Any and all help is much appreciated.
> 
> Hi, I'm having a similar problem though I think I may have a bad TI board. How did you test your encoder? I'd like to test mine as well.

If you have the same encoder me and Clough42 have, then you can take a multimeter and touch the + probe to VCC 5v and the - probe to A and slowly turn the encoder while the encoder is powered. It should at some spots give 5v. Then do the same on b and c legs.

**Comment by Chucksbp** [comment]
Did you re configure, re flash the TI board?? you just cant move to the other input. On code composer right below the resolution it says #define encoder_use_equep1 below it says //#define encoder_use_equep2   just go in and add the // in front of the first line and remove the // from the second line, reprogram and then change your encoder to the other input. if it still doesn't work i would highly suspect you have a TI board problem. Have you also checked for both 3 volt and 5 volt power on both boards. 
  Also are the proper switches uncovered and moved as per instructions??  they are hard to distinguish if they have moved, you may try flicking them back and fort a few times. also di you remove the three jumpers?? Just some thoughts that came to me.

**Comment by MicrowavedBread** [comment]
> 

Yes to all of the above. I changed the port to 2 and checked both encoders. Then back to 1 and checked both encoders again. I guess I'll order a new TI board. Is there a way I can check the board?
My career has gotten me very good a low voltage diagnosis and wiring but once I get to not so visible things on PCB's I'm mostly at a loss.

**Comment by Chucksbp** [comment]
I am sure the board could be diagnosed but I doubt that it could be done with normal DIY tools. Plus you would need some schematics with voltages and possibly traces to follow.
 The boards are relatively inexpensive so unless one could do it yourself it would not be cost effective. I also feel they may be a bit fragile to static charge. they seem to be prone to failure at the encoder input???? On my first attempt i didn't use any static protection I don't know if that led to my failure. I also had a display that either I ruined or it was bad when I received it. All the connections checked out but it never worked. I don't see how that could affect the TI but again I am not an electronics expert. Like you I have no problem with the wiring and following instructions but as far as the software mumbo-jumbo and the fine points of the electronics, NO don't get it!

**Comment by MicrowavedBread** [comment]
> 

Excellent information, thanks for the help.
I'll update when my board gets here and I test it.

**Comment by KevinP8899** [comment]
I'm having the same issue, and I'm fairly certain that the encoder is working. I want to make sure I'm not making a mistake myself before I order another board. I've triple-checked the wiring and tried the second set of pins while reflashing. When testing the encoder with just the (+,-) connected to the board, holding the (-) on the ground and the (+) on any of the 3 signal wires, I get 5.14v (not spinning). When I turn the encoder, the voltage drops significantly. Please tell me if I'm doing this wrong!! When I test the voltage on the board with (- to ground) and (+ to 5v,A,B, or I) , I get 5.14v in the pins (with the encoder disconnected) on both J12 and J13. Is that correct, or should the voltage be different? working with small electronics is very new to me so this is a learning process for sure.

**Comment by Chucksbp** [comment]
There was just another gentlemen above that closed his question he also did all the testing etc. turned out to be a failed encoder. This seems to be a much better built encoder 3 people have had success is repairing problems, I hope you didn't ruin your launch board. I would try an encoder first. I tried testing, i had power on one leg but the second didn't seem correct. I personally believe you need an oscilloscope to properly diagnose the encoder. Here is a link to the encoder from Amazon.  https://www.amazon.com/gp/product/B08RS6M32J/ref=ppx_yo_dt_b_asin_title_o07_s00?ie=UTF8&psc=1

**Comment by KevinP8899** [comment]
Thank you for the link. I'll give it a try. I just don't want to ruin an encoder by putting 5v into something that shouldn't have it. I'm afraid the board is sending power to pins that It shouldn't.

**Comment by Chucksbp** [comment]
Honestly, i understand, I was lucky I guess my ELS worked for an hour or two then suddenly stopped.  That was about a year ago and Launch boards were very hard if not imposable to find. I dint feel that I wanted to try the original encoder on the second set of inputs. I tested it by putting a 5 volt source on the =5 volts and ground then measured from ground to each output, i measured both ac and dc voltage. I only got one wire that i thought maybe correct possibly. I really felt i had a bad encoder. So i ordered the Amazon encoder and reprogramed the board. It worked fine. before that i also had a bad display, don't know if i damaged it or it never did work. 
  I actually ordered a second Clough42 board and have to spare Launch boards. One is programed and tested with all the spare parts except for the motor. I was afraid someday pieces may not be available. I was going to use one for a power down feed on my mill, but went with a different setup for that. Most of the parts are inexpensive, the motor is the largest expense.
   Don't give up! once you get it going you will love it. One thing to check closely are the switches, make sure you work them back and forth a few times, even clean them with some alcohol that sticky tape can give you a false sense of movement.

**Comment by patl1** [comment]
Kevin and Chuck,
I am having the exact symptoms that Kevin described.....5 V on all the pins except the ground line coming out of the Ti board. Is this correct to have 5V on all the lines?. I will test the encoder today by applying the + and - connections to the encoder and look at the V on each of the phase pins (A/B/X). I may decide to order the new encoder from the link as that seemed to work for Kevin, but I don't want to start throwing parts at it if I can isolate which device is the culprit.

**Comment by Chucksbp** [comment]
i cant confirm the reading on the pins, but i can only say it doesn't make sense that there would be voltage at the other pins. 
you may want to configure the other set of pins, do not connect the encoder and measure the voltages. in my mind the other three pins should be inputs to receive the signal from the encoder. 
Have you measured voltages and continuity on the two boards as described in the trouble shooting guide?
 Both of ours worked for a short time before having a problem so we were sure of the integrity of the build.

**Comment by patl1** [comment]
I checked all the connections for continuity as suggested. Both headers are measuring 5V on all pins except ground…no encoder installed and has never been installed on the 2nd header. The phase pins could be a pull down design meaning the encoder pulls them to ground to create the signal….not sure but it is the only way I can think of that would produce 5V without the encoder installed. 

 

I just ordered the amazon encoder and will test it. You should read the 1st review (from “Alex”) on the amazon page for this link. Pretty funny!

**Comment by patl1** [comment]
Is anyone able to check the voltages on the 3 pins (A,B,and I) on the header for the encoder on the Ti board (with the header plug from the encoder unplugged)? This would confirm to me that mine is either correct at 5 V or not. I have ordered a new encoder, but would really like to not shotgun this problem and the information re: the pin voltage would tell me a lot.

**Comment by KevinP8899** [comment]
Pat, 
I had 5v in every pin except the ground on the new board also. I ended up ordering a new encoder and that solved the problem.

**Comment by patl1** [comment]
Thanks Kevin, Really helps. I’ll let you know after I get and try my new encoder on Thursday Pat

**Comment by patl1** [comment]
I got the Amazon encoder suggested in the link above and it worked to produce an rpm signal. Thank you for the assistance and the information on the pin out voltages to help me narrow down the issue.
